src/cpl/nuopc/rof_shr_methods.F90

```diff
@@ -0,0 +1,838 @@
+  use ESMF         , only : operator(-), operator(*) , operator(>=)
+  use ESMF         , only : operator(<=), operator(>), operator(==)
+  use ESMF         , only : ESMF_LOGERR_PASSTHRU, ESMF_LogFoundError, ESMF_LOGMSG_ERROR, ESMF_MAXSTR
+  use ESMF         , only : ESMF_SUCCESS, ESMF_LogWrite, ESMF_LOGMSG_INFO, ESMF_FAILURE
+  use ESMF         , only : ESMF_State, ESMF_StateGet
+  use ESMF         , only : ESMF_Field, ESMF_FieldGet
+  use ESMF         , only : ESMF_GridComp, ESMF_GridCompGet, ESMF_GridCompSet
+  use ESMF         , only : ESMF_GeomType_Flag, ESMF_FieldStatus_Flag
+  use ESMF         , only : ESMF_Mesh, ESMF_MeshGet
+  use ESMF         , only : ESMF_GEOMTYPE_MESH, ESMF_GEOMTYPE_GRID, ESMF_FIELDSTATUS_COMPLETE
+  use ESMF         , only : ESMF_Clock, ESMF_ClockCreate, ESMF_ClockGet, ESMF_ClockSet 
+  use ESMF         , only : ESMF_ClockPrint, ESMF_ClockAdvance 
+  use ESMF         , only : ESMF_Alarm, ESMF_AlarmCreate, ESMF_AlarmGet, ESMF_AlarmSet
+  use ESMF         , only : ESMF_Calendar, ESMF_CALKIND_NOLEAP, ESMF_CALKIND_GREGORIAN
+  use ESMF         , only : ESMF_Time, ESMF_TimeGet, ESMF_TimeSet
+  use ESMF         , only : ESMF_TimeInterval, ESMF_TimeIntervalSet, ESMF_TimeIntervalGet
+  use ESMF         , only : ESMF_VM, ESMF_VMGet, ESMF_VMBroadcast, ESMF_VMGetCurrent
+  use NUOPC        , only : NUOPC_CompAttributeGet
+  use NUOPC_Model  , only : NUOPC_ModelGet
+  use shr_kind_mod , only : r8 => shr_kind_r8, cl=>shr_kind_cl, cs=>shr_kind_cs
+  use shr_sys_mod  , only : shr_sys_abort
+  use shr_file_mod , only : shr_file_setlogunit, shr_file_getLogUnit
+
+  implicit none
+  private
+
+  public  :: memcheck
+  public  :: get_component_instance
+  public  :: set_component_logging
+  public  :: log_clock_advance
+  public  :: state_getscalar
+  public  :: state_setscalar
+  public  :: state_diagnose
+  public  :: alarmInit  
+  public  :: chkerr
+
+  private :: timeInit
+  private :: field_getfldptr
+
+  ! Clock and alarm options
+  character(len=*), private, parameter :: &
+       optNONE           = "none"      , &
+       optNever          = "never"     , &
+       optNSteps         = "nsteps"    , &
+       optNStep          = "nstep"     , &
+       optNSeconds       = "nseconds"  , &
+       optNSecond        = "nsecond"   , &
+       optNMinutes       = "nminutes"  , &
+       optNMinute        = "nminute"   , &
+       optNHours         = "nhours"    , &
+       optNHour          = "nhour"     , &
+       optNDays          = "ndays"     , &
+       optNDay           = "nday"      , &
+       optNMonths        = "nmonths"   , &
+       optNMonth         = "nmonth"    , &
+       optNYears         = "nyears"    , &
+       optNYear          = "nyear"     , &
+       optMonthly        = "monthly"   , &
+       optYearly         = "yearly"    , &
+       optDate           = "date"      , &
+       optIfdays0        = "ifdays0"   
+
+  ! Module data
+  integer, parameter :: SecPerDay = 86400 ! Seconds per day
```

Comment:
Shouldn't this use the SHR_CONST value?

Reply:
I needed an integer not a real (as is the case in shr_const_mod). I could do a conversion - but this integer will not change I don't see a problem for this to stay how it is.